.github/workflows/reusable-workflow.yaml

```diff
@@ -0,0 +1,127 @@
+on:
+    workflow_dispatch:
+    workflow_call:
+      inputs:
+        lib:
+          required: true
+          type: string
+    
+jobs:
+  test-android:
+    runs-on: ubuntu-latest
+    env:
+      BUNDLE_GEMFILE: ${{ github.workspace }}/.github/DangerFiles/Gemfile
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          # We need a sufficient depth or Danger will occasionally run into issues checking which files were modified.
+          fetch-depth: 100
+      - name: Install Dependencies
+        env: 
+          TEST_CREDENTIALS: ${{ secrets.TEST_CREDENTIALS }}
+        run: |
+          ./install.sh
+          echo $TEST_CREDENTIALS > ./shared/test/test_credentials.json
+      - name: Hybrid Dependencies
+        if: ${{ inputs.lib == 'SalesforceHybrid' }}
+        run: |
+          npm install -g cordova
+          cordova telemetry off
+      - name: React Native Dependencies
+        if: ${{ inputs.lib  == 'SalesforceReact' }}
+        run: |
+          npm install -g typescript
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: '21'
+      - name: Setup Android SDK
+        uses: android-actions/setup-android@v3
+      - uses: gradle/actions/setup-gradle@v4
+        with:
+          gradle-version: 8.7
+      - name: Run Lint
+        if: ${{ github.event_name }} == 'pull_request'
+        run: |
+          ./gradlew libs:${{ inputs.lib }}:lint
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: '3.2'
+          bundler-cache: true
+      - name: Report Static Analysis
+        if: ${{ github.event_name }} == 'pull_request'
+        env:
+          DANGER_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          LIB: ${{ inputs.lib }}
+        run: bundle exec danger --dangerfile=.github/DangerFiles/StaticAnalysis.rb --danger_id="${{ inputs.lib }}"
+      - name: Build for Testing
+        if: success() || failure()
+        run: |
+          ./gradlew libs:${{ inputs.lib }}:assembleAndroidTest
+          ./gradlew native:NativeSampleApps:RestExplorer:assembleDebug
+      - uses: 'google-github-actions/auth@v2'
+        if: success() || failure()
+        with:
+          credentials_json: '${{ secrets.GCLOUD_SERVICE_KEY }}'
+      - uses: 'google-github-actions/setup-gcloud@v2'
+        if: success() || failure()
+      - name: Run Tests
+        if: success() || failure()
+        run: |
+          # Most used according to https://gs.statcounter.com/android-version-market-share/mobile-tablet/worldwide
+          DEFAULT_API_VERSION="34" 
+          FULL_API_RANGE="28 29 30 31 32 33 34 35"
```

Comment:
This (along with the Java and Ruby versions above) are why we _really_ need YAML Anchors.  If/when they finally add it we can move this to the top where it can be more easily found and updated when necessary.